tests/components/speedtestdotnet/test_init.py

```diff
@@ -1,79 +1,113 @@
 
     entry = MockConfigEntry(
-        domain=speedtestdotnet.DOMAIN,
+        domain=DOMAIN,
         data={},
+        options={
+            CONF_SERVER_NAME: "Country1 - Sponsor1 - Server1",
+            CONF_SERVER_ID: "1",
+            CONF_SCAN_INTERVAL: 30,
+            CONF_MANUAL: False,
+        },
     )
     entry.add_to_hass(hass)
 
-    with patch("speedtest.Speedtest"), patch(
-        "homeassistant.config_entries.ConfigEntries.async_forward_entry_setup",
-        return_value=True,
-    ) as forward_entry_setup:
-        await hass.config_entries.async_setup(entry.entry_id)
+    await hass.config_entries.async_setup(entry.entry_id)
 
-    assert entry.state is config_entries.ConfigEntryState.LOADED
-    assert forward_entry_setup.mock_calls[0][1] == (
-        entry,
-        "sensor",
-    )
+    assert entry.state == ConfigEntryState.LOADED
+    assert hass.data[DOMAIN]
+    assert hass.services.has_service(DOMAIN, SPEED_TEST_SERVICE)
 
 
-async def test_setup_failed(hass):
+async def test_setup_failed(hass: HomeAssistant, mock_api: MagicMock) -> None:
     """Test SpeedTestDotNet failed due to an error."""
 
     entry = MockConfigEntry(
-        domain=speedtestdotnet.DOMAIN,
-        data={},
+        domain=DOMAIN,
     )
     entry.add_to_hass(hass)
 
-    with patch("speedtest.Speedtest", side_effect=speedtest.ConfigRetrievalError):
+    mock_api.side_effect = speedtest.ConfigRetrievalError
+    await hass.config_entries.async_setup(entry.entry_id)
+    assert entry.state is ConfigEntryState.SETUP_RETRY
 
-        await hass.config_entries.async_setup(entry.entry_id)
 
-    assert entry.state is config_entries.ConfigEntryState.SETUP_RETRY
-
-
-async def test_unload_entry(hass):
+async def test_unload_entry(hass: HomeAssistant) -> None:
     """Test removing SpeedTestDotNet."""
     entry = MockConfigEntry(
-        domain=speedtestdotnet.DOMAIN,
-        data={},
+        domain=DOMAIN,
     )
     entry.add_to_hass(hass)
 
-    with patch("speedtest.Speedtest"):
-        await hass.config_entries.async_setup(entry.entry_id)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
 
     assert await hass.config_entries.async_unload(entry.entry_id)
     await hass.async_block_till_done()
 
-    assert entry.state is config_entries.ConfigEntryState.NOT_LOADED
-    assert speedtestdotnet.DOMAIN not in hass.data
+    assert entry.state is ConfigEntryState.NOT_LOADED
+    assert DOMAIN not in hass.data
+
+
+async def test_server_not_found(hass: HomeAssistant, mock_api: MagicMock) -> None:
+    """Test configured server id is not found."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+    )
+    entry.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+    assert hass.data[DOMAIN]
+
+    mock_api.return_value.get_servers.side_effect = speedtest.NoMatchedServers
+    await hass.data[DOMAIN].async_refresh()
```

Comment:
Please don't interact with integration details like hass.data in the tests. If we want to have the integration do another request of data we can move time forward with our test helper for that.

https://github.com/home-assistant/core/blob/c1b18f48678dd40a333ea05a588be1caf5b2a050/tests/common.py#L363-L366

https://developers.home-assistant.io/docs/development_testing#writing-tests-for-integrations